ingest/expression_files/dense_ingestor.py

```diff
@@ -241,31 +241,34 @@ def transform(self):
         ):
 
             data_arrays.append(all_cell_model)
-        # Represents row as a list
-        for row in self.csv_file_handler:
-            valid_expression_scores, exp_cells = DenseIngestor.filter_expression_scores(
-                row[1:], self.header
-            )
-            exp_scores = DenseIngestor.process_row(valid_expression_scores)
-            gene = row[0]
-            GeneExpression.dev_logger.debug(f"Processing {gene}")
-            if gene in self.gene_names:
-                raise ValueError(f"Duplicate gene: {gene}")
-            self.gene_names[gene] = True
-
-            data_arrays, gene_models, num_processed = self.create_models(
-                exp_cells,
-                exp_scores,
-                gene,
-                None,
-                gene_models,
-                data_arrays,
-                num_processed,
-                False,
-            )
+        if not GeneExpression.is_raw_count_file(
+            self.study_id, self.study_file_id, self.mongo_connection._client
+        ):
+            # Represents row as a list
+            for row in self.csv_file_handler:
+                valid_expression_scores, exp_cells = DenseIngestor.filter_expression_scores(
+                    row[1:], self.header
+                )
+                exp_scores = DenseIngestor.process_row(valid_expression_scores)
+                gene = row[0]
+                GeneExpression.dev_logger.debug(f"Processing {gene}")
```

Comment:
This is a great catch, and I completely agree. I enjoyed having it because it let us know what gene it failed on. I do not think it's necessary now because we know how many models were uploaded at a given time and how many batches where uploaded.